### PR TITLE
Refactor automation

### PR DESCRIPTION
**Pull request by antrix1989**
## Proposed changes

Automation refactoring that includes:
- Shared automation code between Broker and MSAL was moved into common core.
- Added support of different sdks requirements (support of scopes and resources, etc)
- Added support of file pipeline on the simulator (we still keep ui-based pipeline for real device). 

## Type of change

- [ ] Feature work
- [ ] Bug fix
- [ ] Documentation
- [x] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/tests/automation/shared/MSIDAutomationActionConstants.m [hunk 1]
```diff
@@ -41,4 +41,46 @@ @implementation MSIDAutomationActionConstants
...
+    });
+
+    return  path;
+}
+
+#pragma mark - Private
+
++ (NSURL *)cachesDirUrl
+{
+    NSString *simulatorSharedDir = [NSProcessInfo processInfo].environment[@"SIMULATOR_SHARED_RESOURCES_DIRECTORY"];
```
Comment (oldalton) [inline IdentityCore/tests/automation/shared/MSIDAutomationActionConstants.m:81]: Will this work on an iOS device?

File: IdentityCore/tests/automation/ui_app_lib/ui_common/ios/MSIDAutomationRequestViewController.m [hunk 2]
```diff
@@ -1,62 +0,0 @@
...
-// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
-// THE SOFTWARE.
-
-#import "MSIDAutomationRequestViewController.h"
-#import "MSIDAutomationTestRequest.h"
-
-@interface MSIDAutomationRequestViewController ()
```
Comment (oldalton) [inline IdentityCore/tests/automation/ui_app_lib/ui_common/ios/MSIDAutomationRequestViewController.m:27]: Maybe we can support both modes in the automation library:
1. Files based (default)
2. UI based - used for tests that are flaky or don't work with a files based model

Do you see any benefit in supporting both of those models?

File: IdentityCore/tests/automation/ui_tests_lib/MSIDTestConfigurationProvider.m [hunk 3]
```diff
@@ -174,11 +182,30 @@ - (MSIDAutomationTestRequest *)defaultAppRequest:(NSString *)environment
     
     NSString *testEnvironment = environment ? environment : self.wwEnvironment;
     
-    request.requestScopes = [self scopesForEnvironment:testEnvironment type:@"ms_graph"];
-    request.requestResource = [self resourceForEnvironment:testEnvironment type:@"ms_graph"];
-    request.expectedResultScopes = [NSString msidCombinedScopes:request.requestScopes withScopes:[self scopesForEnvironment:testEnvironment type:@"oidc"]];
+    if (self.testsConfig.scopesSupported)
```
Comment (oldalton) [inline IdentityCore/tests/automation/ui_tests_lib/MSIDTestConfigurationProvider.m:185]: Can you provide some context on why this transformation is needed?

File: IdentityCore/tests/automation/ui_app_lib/ui_common/ios/MSIDAutomationRequestViewController.m (hunk 2 above)
Comment (antrix1989) [reply IdentityCore/tests/automation/ui_app_lib/ui_common/ios/MSIDAutomationRequestViewController.m:27]: I don't see the benefit in UI bases pipeline, files communication is faster and has less code.

File: IdentityCore/tests/automation/ui_tests_lib/MSIDTestConfigurationProvider.m (hunk 3 above)
Comment (antrix1989) [reply IdentityCore/tests/automation/ui_tests_lib/MSIDTestConfigurationProvider.m:185]: It is needed for MSAL CPP integration, they don't support scopes.

File: IdentityCore/tests/automation/shared/MSIDAutomationActionConstants.m (hunk 1 above)
Comment (antrix1989) [reply IdentityCore/tests/automation/shared/MSIDAutomationActionConstants.m:81]: Added support of ui pipeline for the real device.

File: IdentityCore/tests/automation/ui_app_lib/ui_common/ios/MSIDAutomationRequestViewController.m (hunk 2 above)
Comment (antrix1989) [reply IdentityCore/tests/automation/ui_app_lib/ui_common/ios/MSIDAutomationRequestViewController.m]: Added support of ui pipeline for the real device.

**Review by oldalton — APPROVED** [review]
Code changes look good to me, but please do take a moment to fill in PR description. It will help others to go back in history and learn why certain changes were made.

**Pull request merged**